### ISSUE TITLE
DEAD flight with AttackGroup task fires all missiles against a single target

### ISSUE THREAD

**Issue by RndName** [dcs-liberation/dcs_liberation#2152]
### Affected versions

5.2.0, Development build

### Description

We set the AttackGroup task for DEAD flights with the property ["expend"]="All" instead of ["expend"]="Auto" like DCS does by default. This causes the AI to fire all available missiles at one single target. This makes the DEAD flight extremly uneffective.

![grafik](https://user-images.githubusercontent.com/6678803/162049953-4cd0f82c-275a-4530-97a1-c68e3c8bad85.png)

Left is the Task in the mission file from Liberation, right is the task as i just removed the task and recreated it. 

![grafik](https://user-images.githubusercontent.com/6678803/162050036-081e397a-7ed5-46db-b795-fe87a5049e22.png)

The expend property maps to "REL QTY". which has these possible values:
![grafik](https://user-images.githubusercontent.com/6678803/162050269-a21068b0-6fb2-4a73-8f71-ad6166b824f7.png)

a short test showed that the AI does what expected if its changed to AUTO.

Solution: Update Pydcs with the correct properties for this Task

### Save game and other files

This mission shows All vs Auto
[attack_group_test.zip](https://github.com/dcs-liberation/dcs_liberation/files/8429837/attack_group_test.zip)

**Comment by DanAlbert** [comment]
How well does "auto" work when the loadout is plain CBUs? That's sort of the "expected" loadout for this mission type.

**Comment by RndName** [comment]
In this example it fires 1 JSOW at each target. I think i've read sth in the dcs forum about the AI calculating the amount of ammunition needed against the given target. but this is highly speculative tbh. I am not sure :D
But Auto is still way better as we normaly have at least two aircraft coming in with a huge loadout.

Also i found out in my testing that the AI is not targeting the Ural Truck standing nearby (same group!). It focuses on the higher values (just my interpretation of what is happening again)

**Comment by DanAlbert** [comment]
My worry is that with non-standoff weapons the AI will take multiple passes over the target dropping one CBU per target and just end up getting killed once the SAM wakes back up. With unguided bombs they really need to do one pass haul ass.

**Comment by RndName** [comment]
With unguided the AI is cheating :D they drop them immediatly at the correct targets.

![grafik](https://user-images.githubusercontent.com/6678803/162061466-ca26f36b-eccb-4c41-813b-b8dfdd83f803.png)

**Comment by DanAlbert** [comment]
Nice, okay, so this improves both behaviors.

**Comment by RndName** [comment]
we can also set attackQty to 1 if we want SEAD and DEAD to only do one attack run at all. We currently have attackQtyLimit disabled so the AI can do what it wans if i understand this parameter correctly

**Comment by Dovahkiin-r** [comment]
> With unguided the AI is cheating :D they drop them immediatly at the correct targets.
> 
> ![grafik](https://user-images.githubusercontent.com/6678803/162061466-ca26f36b-eccb-4c41-813b-b8dfdd83f803.png)

Isn't CBU 103 a guides Bomb? Humans can do that too, beacuse that works like GPS guided bombs...

**Comment by DanAlbert** [comment]
Good catch, yes. The unguided version of that is the 99? 103/105 are WCMD, so that should probably be retested.

**Comment by RndName** [comment]
Ok you were right i checked it with CBU-87s 8 per Aircraft, 2 Aircrafts per package.
![grafik](https://user-images.githubusercontent.com/6678803/162401699-fd1de688-3a92-42c3-b175-bb50d2bef00b.png)

Results Unguided CBU-87 (8x) and MK-82 (4x each)
- All and Group Attack: they fire all CBUs against one target (both flights the same): 1/8 killed in 1 run
- **All and no Group Attack**: they fire all CBUs at once but against different targets: 2/8 killed in 1 run
- Auto and Group Attack: they fire 2 CBUs at once at the same target both. Even with AttackQty set to 1 they are doing multiple attack runs until theyre empty: 2/8 killed in 2 runs
- Auto and no Group Attack: hey fire 2 CBUs at once at different targets. Even with AttackQty set to 1 they are doing multiple attack runs until theyre empty 4/8 killed in 2 runs

Results Guided JSOW-A (4x) and JDAMs tested. Results are with JDAMS (4x each)
- All and Group Attack: they fire all JSOWs against one target at once. Both flights target the same target. 1/8 killed
- All and no Group Attack: they fire all JDAMs against one target at once but both flights have a different target  2/8 killed
- **Auto and Group Attack**: JDAMs: 4/8 killed, JSOW:
- **Auto and no Group Attack**: 8/8 killed
All and Auto only did 1 attack run!

With GroupAttack enabled they do an immense overkill on a single target all the time. If they are set to All then i would definetly disable GroupAttack as otherwise they only attack 1 single target with all weapons they have which imo is totally useless.

When set to Auto the AI is a bit more intelligent and it is debatable if we choose Group Attack or not but i would highly recommend disabling it either way. 

Type | Armament | Target | Expend | Group Attack | Result | AttackRuns
-- | -- | -- | -- | -- | -- | --
Unguided | MK-82 4x2 | 8xZU-23 | All | on | 1/8 | 1
Unguided | MK-82 4x2 | 8xZU-23 | All | off | 2/8 | 1
Unguided | MK-82 4x2 | 8xZU-23 | Auto | on | 2/8 | 2
Unguided | MK-82 4x2 | 8xZU-23 | Auto | off | 4/8 | 2
Guided | GBU-38 4x2 | 8xZU-23 | All | on | 1/8 | 1
Guided | AGM-154A 4x2 | SA-11 Site | All | on | 2/7 | 1
Guided | GBU-38 4x2 | 8xZU-23 | All | off | 2/8 | 1
Guided | AGM-154A 4x2 | SA-11 Site  | All | off | 5/7| 1
Guided | GBU-38 4x2 | 8xZU-23 | Auto | on | 5/8 | 1
Guided | AGM-154A 4x2 | SA-11 Site  | Auto | on | 5/7 | 1
Guided | GBU-38 4x2 | 8xZU-23 | Auto | off | 8/8 | 1
Guided | AGM-154A 4x2 | SA-11 Site  | Auto | off | 7/7| 1


On Auto they are doing multiple attack runs even if AttackQty limit is enabled and set to 1. Found a thread with the same issue on the forum https://forum.dcs.world/topic/274672-how-does-max-attack-quantity-works/

The problem is to identitfy if the Flight uses Unguided or Guided munition.

So to sum up all my testing i would recommend: expend=Auto, groupAttack=False. Downside is that unguided munition can cause the AI to do multiple Attack runs. To prevent this we would have to check if the AI is using unguided or guided munition or set expend=All, groupAttack=False with the downsides of dumb AI :D Still i would highly recommend groupAttack to turn off.

Test mission: [attack_group_test.zip](https://github.com/dcs-liberation/dcs_liberation/files/8450680/attack_group_test.zip)

**Comment by RndName** [comment]
I overall had really bad results using GroupAttack enabled: With AGM-154A the AI shoot all 8 missiles at the SR when i tried testing against an SA-2 Site. With GroupAttack disabled all 8 missiles were spread a bit and the whole Site was destroyed.
In Theory i would say the GroupAttack could be good but the implementation seems to be super dumb and i would totally prefer disabling it. In all my tests the overall chance of destroying the highest threats was without using GroupAttack.

Regarding the All/Auto Setting the results were quite even when using the JSOW.
Just one side note there as well: DCS Mission editor would not allow us to set Expend=All when using WeaponType Auto. It is just disabled and defaults to Auto as well. Only way to set it through ME is to change Weapon type to guided or unguided

**Comment by RndName** [comment]
Also this could be interesting for #270 as there my be specific Attack Task profiles for each Aircraft. The default value we choose is always a tradeoff and does not work equally for all missions and aircrafts.

**Comment by DanAlbert** [comment]
As long as the selected target is one of the radars, group attack sounds like the better option? Best chance of survival and best chance of removing the threat. Group attack disabled might kill more units, but that doesn't matter. The goal is to make the SAM inoperable, not to wipe it completely off the map. Launchers with no radars aren't any more of a threat than a ural. The AI commander won't even bother planning SEAD/DEAD against them.

The exception where that won't work as well is groups with TELARs or with multiple SRs, but those aren't the common case.

**Comment by RndName** [comment]
> As long as the selected target is one of the radars, group attack sounds like the better option? Best chance of survival and best chance of removing the threat. Group attack disabled might kill more units, but that doesn't matter. The goal is to make the SAM inoperable, not to wipe it completely off the map. Launchers with no radars aren't any more of a threat than a ural. The AI commander won't even bother planning SEAD/DEAD against them.
> 
> The exception where that won't work as well is groups with TELARs or with multiple SRs, but those aren't the common case.

Yeah i tried a bit of testing with Auto and GroupAttack on. Works quite nice even against an SA-11 with only Telars. The AI does a quite okay job in setting the target priority: they killed all the Launchers for the SA-11 Site. For the SA-2 Site they just focused the SR and TR. Really good overall effects.

So maybe Auto and GroupAttack on is the way to go.

Same thing for Strike and BAI. WDYT, Strike: GroupAttack on, BAI GroupAttack off? Strike has the focus of destroying the specific targets so GroupAttack does a good job even if its an overkill. For BAI i think its better to be off as there we want more kills than ensuring it is 100% killed or am i wrong?

**Comment by DanAlbert** [comment]
Makes sense for BAI but I suspect we can't do much for strike since each target is a distinct _task_. Afaict the AI cannot plan ahead.

**Comment by RndName** [comment]
> but I suspect we can't do much for strike

yes we can not set the priority to the AI in regard to the targets but the Bombing task also has the GroupAttack param so at least all pilots attack the targets :)
Gonna update the PR accordingly.

For the future when we come to the point of mission specific flight plans it could also be of interest to add customization possibilites to these task params from the UI. but thats for the future :D

**Comment by Starfire13** [comment]
Attaching save file with kamikaze SEAD flights as Rndname requested. There are 2 packages of 8 Hornets each attacking two different S-300 sites, as well as a whole bunch of other flights attacking a variety of different SAM sites. On the whole, they don't do very well.

![Capture](https://user-images.githubusercontent.com/72491792/163540917-043aa1b0-24c5-48e0-9d0a-f8fefe4e7408.JPG)


[Vegas Nerve #3776.zip](https://github.com/dcs-liberation/dcs_liberation/files/8494663/Vegas.Nerve.3776.zip)
